src/polkadot/fellows.ts

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+import { ApiPromise, WsProvider } from "@polkadot/api";
+
+import { ActionLogger, TeamApi } from "../github/types";
+
+type FellowData = { address: string; rank: number };
+
+export class PolkadotFellows implements TeamApi {
+  private fellowsCache: Map<string, number> = new Map<string, number>();
+
+  constructor(private readonly logger: ActionLogger) {}
+
+  async fetchAllFellows(): Promise<Map<string, number>> {
+    let api: ApiPromise;
+    this.logger.debug("Connecting to collective parachain");
+    // we connect to the collective rpc node
+    const wsProvider = new WsProvider("wss://polkadot-collectives-rpc.polkadot.io");
+    api = await ApiPromise.create({ provider: wsProvider });
+    try {
+      // We fetch all the members
+      const membersObj = await api.query.fellowshipCollective.members.entries();
+
+      // We iterate over the fellow data and convert them into usable values
+      const fellows: FellowData[] = [];
+      for (const [key, rank] of membersObj) {
+        // @ts-ignore
+        const [address]: [string] = key.toHuman();
+        fellows.push({ address, ...(rank.toHuman() as object) } as FellowData);
+      }
+      this.logger.debug(JSON.stringify(fellows));
+
+      // Once we obtained this information, we disconnect this api.
+      await api.disconnect();
+
+      this.logger.debug("Connecting to relay parachain.");
+      // We connect to the relay chain
+      api = await ApiPromise.create({ provider: new WsProvider("wss://rpc.polkadot.io") });
+
+      // We iterate over the different members and extract their data
+      const users: Map<string, number> = new Map<string, number>();
+      for (const fellow of fellows) {
+        this.logger.debug(`Fetching identity of '${fellow.address}', rank: ${fellow.rank}`);
+        const fellowData = (await api.query.identity.identityOf(fellow.address)).toHuman() as
```

Comment:
I feel like this deserves a semaphore, so it could be queried in parallel.

Reply:
Do you know if there is a limit to stop abuse? That was my main concern.

The current execution takes about 2s, which given the fact that this is an action, is not really a problem (the whole action is executed in about 5s).

Reply:
> Do you know if there is a limit to stop abuse? That was my main concern.

Could you elaborate?

Reply:
Same problem that we have with GitHub, which has a rate limit.

I assume that there is some kind of rate limit when using Polkadot-JS (else what stop people from DOSing it?), and, because I'm fetching 60 accounts, I found it better to do it in order.

Reply:
Maybe I'm not understanding you. Could you please elaborate on what you had in mind?

Reply:
I think we're speaking of the same thing here. I suggest adding a semaphore, so requests would be processed in parallel, but with a limit on how much parallel requests to make. 

In `gitspiegel`, I've used [semaphore-promise](https://github.com/Granjow/semaphore-promise) library for that.  
Check out the usage in [core.ts](https://github.com/paritytech/gitspiegel/blob/master/src/core.ts#L22C7-L22C25)